core/src/main/java/org/mule/api/registry/Registry.java

```diff
@@ -116,8 +117,10 @@
      * @param metadata an implementation specific argument that can be passed into the method
      * @throws RegistrationException if there is a problem unregistering the object. Typically this will be because
      * the object's lifecycle threw an exception
+     * @deprecated as of 3.7.0. Use {@link #unregisterObject(String)} instead
      */
-    void unregisterObject(String key, Object metadata) throws RegistrationException;
+    @Deprecated
+    Object unregisterObject(String key, Object metadata) throws RegistrationException;
```

Comment:
If a method changes the signature, then is a new method, why do you need to deprecate it?